tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorValidationTests.cs

```diff
@@ -59,6 +59,75 @@ public void Should_not_throw_the_exception_when_a_child_expression_is_called_fro
             Execute(CreateWhereQuery(expr));
         }
 
+        [Fact]
+        public void Should_not_throw_the_exception_when_a_child_expression_is_called_from_nested_locals()
+        {
+            Setup();
+
+            var local = new List<int> { 1, 2 };
+            var local2 = new List<bool> { true, false };
+
+            Expression<Func<TestObject, bool>> expr = (a) => a.Collection1.Any(b => local.Any(c => b.Collection2.Contains(c)));
+            AssertWhere(expr, "{ $match : { 'Collection2' : { '$in' : [1, 2] } } }");
+
+            expr = (a) => a.Collection1.Any(b => b.Collection1.Any(c => local.Any(d => c.Collection2.Contains(d))));
```

Comment:
This linq query generates the same mongo query as in the line above which is strange. It doesn't look like the behavior related to this ticket, but I would create a new ticket to investigate the reasons of it.